cachito/web/models.py

```diff
@@ -566,6 +566,73 @@ def _validate_request_package_configs(request_kwargs, pkg_managers_names):
                 _validate_configuration_path_value(pkg_manager, "requirements_build_files", path)
 
+    _validate_package_manager_exclusivity(
+        pkg_managers_names, packages_configs, PackageManager.LOCALLY_EXCLUSIVE
+    )
+
+
+def _validate_package_manager_exclusivity(pkg_manager_names, package_configs, mutually_exclusive):
+    """
+    Ensure that no package gets processed by two or more mutually exclusive package managers.
+
+    Note: git-submodule is a special case, because we always fetch all submodules. Therefore
+    we do not know which subpaths are actually submodules prior to processing the request, and
+    we have to assume that any non-root path is a submodule.
+
+    :param list pkg_manager_names: the list of package manager names for the request
+    :param dict package_configs: the "packages" parameter in a request
+    :param list mutually_exclusive: see `PackageManager.LOCALLY_EXCLUSIVE`
+    :raises ValidationError: if the package configuration has conflicting paths (even implicitly)
+    """
+    mutually_exclusive = set(mutually_exclusive)
+
+    pkg_manager_paths = {
+        pkg_manager: set(
+            os.path.normpath(pkg_cfg.get("path", "."))
+            for pkg_cfg in package_configs.get(pkg_manager, [{}])
+        )
+        for pkg_manager in pkg_manager_names
+        if pkg_manager != "git-submodule"
+    }
+
+    if "git-submodule" in pkg_manager_names:
+        _validate_gitsubmodule_exclusivity(pkg_manager_paths, mutually_exclusive)
+
+    for a in pkg_manager_paths:
+        # Check all package manager pairs
+        for b in pkg_manager_paths:
+            if a == b or not ((a, b) in mutually_exclusive or (b, a) in mutually_exclusive):
+                continue
+
+            conflicting_paths = pkg_manager_paths[a] & pkg_manager_paths[b]
+            if conflicting_paths:
+                msg = (
+                    f"The following paths cannot be processed by both '{a}' and '{b}': "
+                    f"{', '.join(sorted(conflicting_paths))}"
+                )
+                raise ValidationError(msg)
+
+
+def _validate_gitsubmodule_exclusivity(pkg_manager_paths, mutually_exclusive):
+    """
+    Validate exclusivity of git-submodule with other package managers.
+
+    :param dict pkg_manager_paths: mapping of package managers and their paths in a request
+    :param set mutually_exclusive: see `PackageManager.LOCALLY_EXCLUSIVE` (converted to set)
+    :raises ValidationError: if any package manager conflicts with git-submodule
+    """
+    for pkg_manager, paths in pkg_manager_paths.items():
+        a, b = pkg_manager, "git-submodule"
+        if not ((a, b) in mutually_exclusive or (b, a) in mutually_exclusive):
+            continue
+
+        if any(path != "." for path in paths):
+            msg = (
+                f"Cannot process non-root packages with '{pkg_manager}' "
+                "when 'git-submodule' is also set"
+            )
+            raise ValidationError(msg)
```

Comment:
Do we want this? Could come in handy if we decide to disallow gomod+gitsubmodule after #305, but maybe this check should instead be done by a worker, when we know that the subpath is actually a git submodule

cc @athos-ribeiro @arewm

Reply:
Even though it would make processing more difficult, I think that the biggest issue between git submodules and other package managers is when the paths coincide. If that is deemed too hard to implement, however, this seems like a reasonable approximation.

Reply:
+1. We can start with this and  pursue an algorithmic approach  based on how users interact with the feature.